server/lib/notifications/email.ts

```diff
@@ -48,6 +48,11 @@ export const notify = {
       CollectiveId: options?.collective?.id || activity.CollectiveId,
     });
 
+    if (!activity.data.recipientName) {
+      const userCollective = await models.Collective.findByPk(user.CollectiveId);
+      activity.data.recipientName = userCollective.name;
```

Comment:
To add support for legal name:

```suggestion
      activity.data.recipientName = userCollective.name || userCollective.legalName;
```